datafusion/core/src/datasource/data_source.rs

```diff
@@ -62,4 +62,8 @@ pub trait FileSource: Send + Sync {
     fn fmt_extra(&self, _t: DisplayFormatType, _f: &mut Formatter) -> fmt::Result {
         Ok(())
     }
+    /// Check if repartition is supported
+    fn supports_repartition(&self, config: &FileScanConfig) -> bool {
+        !(config.file_compression_type.is_compressed() || config.new_lines_in_values)
+    }
```

Comment:
I think it would be better we did not provide a default implementation for  `supports_repartition`

Assuming that a file source can repartition based on its compression and newlines seems very specific to CSV.

```suggestion
    /// Check if repartition is supported
    fn supports_repartition(&self, config: &FileScanConfig) -> bool;
```

Reply:
I will try to include it in my next PR, which should be tomorrow hopefully.

just for clarity do you want to implement something like this for other supported file format
```
fn supports_repartition(&self, config: &FileScanConfig) -> bool {
        !(config.file_compression_type.is_compressed())
    }
  ``` 
 
  and original for csv.

Reply:
No worries! Thanks -- I pushed a change to this branch -- hopefully that was ok

Reply:
it is good.

Reply:
(let's just hope I haven't broken the CI checks. 😆 😭 )

Reply:
🤞